### PR TITLE
Fix error when searching for users with symbols (#416)

### LINKED ISSUE #416: User search: Cannot search for e-mail addresses containing "+"
Attempting to search by e-mail address for any string containing "+" throws an error.

Example request-id: 8a7061be725a464c9e6c3c3f3b8dfaa6-SF+Rg9h6

### PR DESCRIPTION
**Pull request by huntertur**
Fixes #416 

~~Double-backslash prevents the error case you brought up in the issue.~~

### CODE REVIEW

File: weasyl/moderation.py [hunk 1]
```diff
@@ -364,11 +364,10 @@ def finduser(targetid, username, email, dateafter, datebefore, excludesuspended,
     if targetid:
         q = q.where(lo.c.userid == targetid)
     elif username:
-        q = q.where(lo.c.login_name.op('~')(username))
+        q = q.where(lo.c.login_name.op('ilike')('%%%s%%' % username.replace('\\', r'\\')))
```
Comment (charmander) [inline weasyl/moderation.py:367]:
```suggestion
        q = q.where(lo.c.login_name.ilike(f"%{username}%", escape=""))
```

File: weasyl/moderation.py [hunk 2]
```diff
@@ -364,11 +364,10 @@ def finduser(targetid, username, email, dateafter, datebefore, excludesuspended,
...
     elif email:
         q = q.where(sa.or_(
-            lo.c.email.op('~')(email),
-            lo.c.email.op('ilike')('%%%s%%' % email),
+            lo.c.email.op('ilike')('%%%s%%' % email.replace('\\', r'\\')),
```
Comment (charmander) [inline weasyl/moderation.py:370]:
```suggestion
            lo.c.email.ilike(f"%{email}%", escape=""),
```

**Review by charmander — COMMENTED** [review]
readability improvement?

File: weasyl/moderation.py (hunk 1 above)
Comment (philpem) [inline weasyl/moderation.py:367]: Are we on a recent enough python to use f-strings e.g. `f"%{username_r}%"` here, and break out the `username.replace` into a string assignment above?

Comment (charmander) [reply weasyl/moderation.py:367]: Actually, this doesn’t work with SQLAlchemy >:(

Comment (huntertur) [reply weasyl/moderation.py]: Yeah, f-strings are 3.6, and the Dockerfile is for 3.10

**Review by charmander — APPROVED** [review]

**Pull request merged**